test/cluster-isolation/mzcompose.py

```diff
@@ -0,0 +1,270 @@
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable, Optional
+
+from materialize import spawn
+from materialize.mzcompose import Composition
+from materialize.mzcompose.services import (
+    Computed,
+    Kafka,
+    Materialized,
+    SchemaRegistry,
+    Testdrive,
+    Zookeeper,
+)
+
+SERVICES = [
+    Zookeeper(),
+    Kafka(),
+    SchemaRegistry(),
+    Computed(
+        name="computed_1_1",
+        options="--workers 2 --processes 2 --process 0 computed_1_1:2102 computed_1_2:2102 --storage-addr materialized:2101",
+        ports=[2100, 2102],
+    ),
+    Computed(
+        name="computed_1_2",
+        options="--workers 2 --processes 2 --process 1 computed_1_1:2102 computed_1_2:2102 --storage-addr materialized:2101",
+        ports=[2100, 2102],
+    ),
+    Computed(
+        name="computed_2_1",
+        options="--workers 2 --processes 2 --process 0 computed_2_1:2102 computed_2_2:2102 --storage-addr materialized:2101",
+        ports=[2100, 2102],
+    ),
+    Computed(
+        name="computed_2_2",
+        options="--workers 2 --processes 2 --process 1 computed_2_1:2102 computed_2_2:2102 --storage-addr materialized:2101",
+        ports=[2100, 2102],
+    ),
+    Materialized(extra_ports=[2101]),
+    Testdrive(
+        volumes=[
+            "mzdata:/share/mzdata",
+            "tmp:/share/tmp",
+        ],
+        no_reset=True,
+        default_timeout="5s",
+    ),
+]
+
+
+@dataclass
+class Disruption:
+    name: str
+    disruption: Callable
+
+
+disruptions = [
+    Disruption(
+        name="pause-one-computed",
+        disruption=lambda c, _: c.pause("computed_1_1"),
+    ),
+    Disruption(
+        name="kill-all-computed",
+        disruption=lambda c, _: c.kill("computed_1_1", "computed_1_2"),
+    ),
+    Disruption(
+        name="panic-in-materialized-view",
+        disruption=lambda c, id: c.testdrive(
+            """
+> SET cluster=cluster1
+
+> CREATE TABLE panic_table (panic TEXT)
+
+> CREATE MATERIALIZED VIEW panic_view AS SELECT mz_internal.mz_panic(panic) FROM panic_table;
+
+> INSERT INTO panic_table VALUES ('panic')
+""",
+        ),
+    ),
+    Disruption(
+        name="pause-in-aterialized-view",
+        disruption=lambda c, id: c.testdrive(
+            """
+> SET cluster=cluster1
+
+> CREATE TABLE sleep_table (sleep INTEGER);
+
+> CREATE MATERIALIZED VIEW sleep_view AS SELECT mz_internal.mz_sleep(sleep) FROM sleep_table;
+
+> INSERT INTO sleep_table VALUES (1200)
+""",
+        ),
+    ),
+    Disruption(
+        name="drop-cluster",
+        disruption=lambda c, id: c.testdrive(
+            """
+> DROP CLUSTER cluster1
+""",
+        ),
+    ),
+]
+
+# computed: If one cluster has panicked during INSERT ... SELECT, other clusters hang when inserting #12251
```

Comment:
I think the current diagnosis is that ADAPTER hangs awaiting the response from the first compute instance. The other compute instances are not hung, they just aren't being probed by ADAPTER.

Reply:
Yes, sorry, will update the issue title and the comment appropriately.